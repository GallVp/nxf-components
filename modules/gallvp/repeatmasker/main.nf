process REPEATMASKER {
    tag "$meta.id"
    label 'process_high'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/repeatmasker:4.1.5--pl5321hdfd78af_0':
        'biocontainers/repeatmasker:4.1.5--pl5321hdfd78af_0' }"

    input:
    tuple val(meta), path(fasta)
    path(lib)

    output:
    tuple val(meta), path("${prefix}.masked")   , emit: masked
    tuple val(meta), path("${prefix}.out")      , emit: out
    tuple val(meta), path("${prefix}.tbl")      , emit: tbl
    path "versions.yml"                         , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args    = task.ext.args     ?: ''
    prefix      = task.ext.prefix   ?: "${meta.id}"
    def lib_arg = lib               ? "-lib $lib"   : ''
    """
    RepeatMasker \\
        $lib_arg \\
        -pa ${task.cpus} \\
        -dir ${prefix} \\
        ${args} \\
        ${fasta}

    mv $prefix/${fasta}.masked  ${prefix}.masked
    mv $prefix/${fasta}.out     ${prefix}.out
    mv $prefix/${fasta}.tbl     ${prefix}.tbl

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        repeatmasker: \$(RepeatMasker -v | sed 's/RepeatMasker version //1')
    END_VERSIONS
    """

    stub:
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.masked
    touch ${prefix}.out
    touch ${prefix}.tbl

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        repeatmasker: \$(RepeatMasker -v | sed 's/RepeatMasker version //1')
    END_VERSIONS
    """
}
