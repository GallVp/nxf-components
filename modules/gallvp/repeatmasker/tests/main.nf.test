nextflow_process {

    name "Test Process REPEATMASKER"
    script "../main.nf"
    process "REPEATMASKER"
    config "./nextflow.config"

    tag "modules"
    tag "modules_gallvp"
    tag "repeatmasker"

    test("sarscov2 - genome - fasta") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                ]
                input[1] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.masked,
                    process.out.out,
                    process.out.versions
                    ).match() },
                { assert file(process.out.tbl[0][1]).text.contains('run with rmblastn') }
            )
        }

    }

    test("sarscov2 - genome - fasta - stub") {

        options '-stub'

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                ]
                input[1] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
