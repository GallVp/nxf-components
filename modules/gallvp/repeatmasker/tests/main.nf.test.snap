{
    "sarscov2 - genome - fasta": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.masked:md5,c0eb8dd958ce3b4b1fdc7fcb6b0d5161"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.out:md5,8610cb2b8d87356bf2ab0a895c065752"
                ]
            ],
            [
                "versions.yml:md5,f22d471f5aed51b369cc4295ad1f7bc0"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-25T12:17:18.863465"
    },
    "sarscov2 - genome - fasta - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.masked:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tbl:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    "versions.yml:md5,f22d471f5aed51b369cc4295ad1f7bc0"
                ],
                "masked": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.masked:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "out": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "tbl": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tbl:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,f22d471f5aed51b369cc4295ad1f7bc0"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-25T11:01:52.492661"
    }
}